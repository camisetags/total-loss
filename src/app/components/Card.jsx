import React from 'react';
import PropTypes from 'prop-types';

const Card = props => (
  <div className="card card-center">
    <h1>{props.content.name}</h1>
    <p>{props.content.description}</p>
    <hr />
    {props.children}
  </div>
);

Card.propTypes = {
  content: PropTypes.shape({
    name: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
  }).isRequired,
  children: PropTypes.element.isRequired,
};

export default Card;
