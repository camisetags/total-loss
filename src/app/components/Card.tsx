import * as React from 'react';

type CardProps = {
  content: {
    name: string,
    description: string,
  },
  children: React.ReactChild,
};

const Card = (props: CardProps) => (
  <div className="card card-center">
    <h1>{props.content.name}</h1>
    <p>{props.content.description}</p>
    <hr />
    {props.children}
  </div>
);

export default Card;
