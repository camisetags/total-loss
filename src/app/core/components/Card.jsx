import React from 'react';
import PropTypes from 'prop-types';

const Card = props => (
  <div className="card card-center">
    <h1>{props.content.title}</h1>
    <p>{props.content.description}</p>
  </div>
);

Card.propTypes = {
  content: PropTypes.shape({
    title: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
  }).isRequired,
};

export default Card;
