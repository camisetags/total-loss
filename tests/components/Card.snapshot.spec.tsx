import * as React from 'react';
import * as renderer from 'react-test-renderer';
import Card from '../../src/app/components/Card';

describe('Card component', () => {
  const cardData = {
    content: {
      name: 'Testing card',
      description: 'Some testing card',
    },
    children: <div> Testing node </div>,
  };

  it('renders correctly', () => {
    const rendered = renderer.create(<Card {...cardData} />);
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});
