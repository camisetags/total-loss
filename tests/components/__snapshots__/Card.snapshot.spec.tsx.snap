// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component renders correctly 1`] = `
<div
  className="card card-center"
>
  <h1>
    Testing card
  </h1>
  <p>
    Some testing card
  </p>
  <hr />
  <div>
     Testing node 
  </div>
</div>
`;
